components/org.wso2.carbon.identity.conditional.auth.functions.elk/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/elk/CallElasticFunctionImpl.java

```diff
@@ -0,0 +1,200 @@
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.ContentTypes.TYPE_APPLICATION_JSON;
+import static org.wso2.carbon.identity.conditional.auth.functions.common.utils.Constants.*;
+
+/**
+ * Implementation of the {@link CallElasticFunction}
+ */
+public class CallElasticFunctionImpl extends AbstractElasticHelper implements CallElasticFunction {
+
+    private static final Log LOG = LogFactory.getLog(CallElasticFunctionImpl.class);
+
+    private static final ElasticConfigProvider elasticConfigProvider = ElasticConfigProvider.getInstance();
+
+    public CallElasticFunctionImpl() {
+
+        super();
+    }
+
+    @Override
+    public void callElastic(Map<String, String> params, Map<String, Object> eventHandlers) {
+
+        AsyncProcess asyncProcess = new AsyncProcess((authenticationContext, asyncReturn) -> {
+
+            try {
+                String tenantDomain = authenticationContext.getTenantDomain();
+                String targetHostUrl = CommonUtils.getConnectorConfig(ElasticAnalyticsEngineConfigImpl.RECEIVER,
+                        tenantDomain);
+                if (targetHostUrl == null) {
+                    throw new FrameworkException("Elasticsearch host cannot be found.");
+                }
+
+                HttpPost request = new HttpPost(elasticConfigProvider.getElasticSearchUrl(targetHostUrl,params));
+
+                request.setHeader(ACCEPT, TYPE_APPLICATION_JSON);
+                request.setHeader(CONTENT_TYPE, TYPE_APPLICATION_JSON);
+                handleAuthentication(request, authenticationContext.getTenantDomain());
+
+                String query = elasticConfigProvider.getQuery(params);
+                request.setEntity(new StringEntity(query, StandardCharsets.UTF_8));
+
+                String[] targetHostUrls = targetHostUrl.split(";");
+
+                HttpHost[] targetHosts = new HttpHost[targetHostUrls.length];
+
+                for (int i = 0; i < targetHostUrls.length; i++) {
+                    URL hostUrl = new URL(targetHostUrls[i]);
+                    targetHosts[i] = new HttpHost(hostUrl.getHost(), hostUrl.getPort(), hostUrl.getProtocol());
+                }
+
+                CloseableHttpAsyncClient client = HttpClientManager.getInstance().getClient(tenantDomain);
+
+                AtomicInteger requestAtomicInteger = new AtomicInteger(targetHosts.length);
+
+                for (final HttpHost targetHost : targetHosts) {
+                    client.execute(targetHost, request, new FutureCallback<HttpResponse>() {
+
+                        @Override
+                        public void completed(final HttpResponse response) {
+
+                            int responseCode = response.getStatusLine().getStatusCode();
+                            try {
+                                Map<String, Object> responseMap = new HashMap<>();
+                                LOG.error(responseCode);
```

Comment:
Why is the response code always an error log?

Reply:
Resolved in https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/110